src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Toolkit.RuntimeTests.Extensions;
+using Uno.Toolkit.RuntimeTests.Helpers;
+using Uno.Toolkit.UI;
+using Uno.UI.RuntimeTests;
+#if IS_WINUI
+using Microsoft.UI.Xaml.Controls;
+#else
+using Windows.UI.Xaml.Controls;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests
+{
+	[TestClass]
+	[RunsOnUIThread]
+	internal class TabBarTests
+	{
+		[TestMethod]
+		[DataRow(new int[0], 0)]
+		[DataRow(new[] { 1 }, 1)]
+		[DataRow(new[] { 1, 1 }, 1)]
+		[DataRow(new[] { 1, 2 }, 2)]
+		[Ignore("Test ignored until we can simulate clicks")]
```

Comment:
You can cheat this by reflection. ~~Just set a bp on IsSelected.set and Tap/Click, compare their call stack for a common and appropriate method, and just call it.~~
Since you are using a combo of (TabBar as ItemsControl + TabBarItem as SelectorItem), the ideal point of entry would be TabBarItem.ExecuteTap()

Reply:
you can even rename `ToggleButtonExtensions` to `AutomationExtensions` and add your method there.

Reply:
i could also just make toolkit's internals visible to the runtimetest assembly, like we do for uno: https://github.com/unoplatform/uno/blob/3cb7e5f97a9226ba3d0b53756e61a0d94832169c/src/Uno.UI/AssemblyInfo.cs#L12